parser.go

```diff
@@ -161,7 +161,7 @@ func readKeyName(delimiters string, in []byte) (string, int, error) {
 	}
 
 	endIdx = strings.IndexAny(line, delimiters)
-	if endIdx < 0 {
+	if endIdx <= 0 {
 		return "", -1, ErrDelimiterNotFound{line}
```

Comment:
I think we need to return a different error here, because `endIdx == 0` should be `ErrKeyEmptyName` not `ErrDelimiterNotFound`.

Reply:
Makes sense to have a new error for that. I already added it to the fixup commit.